dev/com.ibm.ws.st.liberty.buildplugin.integration/src/com/ibm/ws/st/liberty/buildplugin/integration/manager/internal/AbstractLibertyManager.java

```diff
@@ -923,8 +923,16 @@ private void ensureServersStopped(List<IServer> serverList, IProgressMonitor mon
 
                 Trace.trace(Trace.INFO, "Stopping server: " + server.getName());
                 wsServer.getWebSphereServerBehaviour().stop(true, monitor);
-                if (IServer.STATE_STOPPED != server.getServerState()) {
-                    throw new Exception("Failed to stop server: " + server.getName());
+                wsServer.getWebSphereServerBehaviour().waitForServerStop(10000);
+                long endTime = System.currentTimeMillis() + 10000; // Give it a 10 second time-out
```

Comment:
Should it use the IServer.getStopTimeout() value here?

Reply:
Yes, it is a good idea to use it.